src/cms/views/events/event_view.py

```diff
@@ -191,7 +191,9 @@ def post(self, request, **kwargs):
             if event_form.cleaned_data.get("is_recurring"):
                 # If event is recurring, save recurrence rule
                 event_form.instance.recurrence_rule = recurrence_rule_form.save()
-            elif event_form.fields["is_recurring"].has_changed():
+            elif event_instance and event_form.fields["is_recurring"].has_changed(
+                event_instance.is_recurring, event_form.cleaned_data.get("is_recurring")
+            ):
```

Comment:
```suggestion
            elif event_instance and event_instance.recurrence_rule:
```
Maybe, we can simplify this? If I understand it correctly, we want to delete the recurrence rule object in this case:
- The event exists
- The event was recurring before
- The event should not be recurring anymore

Since the last condition is already checked by the fact that the first if case `event_form.cleaned_data.get("is_recurring")` was not matched, I think we can just check the first two conditions instead of relying on the Django functionality `has_changed`... or am I missing something here?

Reply:
Sorry for taking so long to respond.

You're right, my solution checked the change more explicitly but as I mentioned already above, I was not very happy with my solution anyways.
Since we already know through the first ` if event_form.cleaned_data.get("is_recurring")` which evaluated to false that there should be no recurrence now, we implicitly know that there was a change if the object still exists.